src/type_checker.cpp

```diff
@@ -38,6 +38,18 @@ bool IsInBodyOf(BodyType type) {
                      [type](auto&& t) { return t == type; });
 }
 
+/// @note Add prefix to struct and union type id to avoid same name with
+/// different types in type table.
+std::string RecordTypeId(const std::string& id, std::unique_ptr<Type> type) {
+  if (type->IsStruct()) {
```

Comment:
How about renaming this function to `MangleToRecordTypeId` to better convey its purpose?
```suggestion
/// @note struct and union type id should be mangled when adding/looking up from the type table to avoid same name but different types.
std::string MangleToRecordTypeId(const std::string& id, std::unique_ptr<Type> type) {
  // We simply prefix them with their record kind.
  if (type->IsStruct()) {
```